src/client/classes/Filter.ts

```diff
@@ -0,0 +1,14 @@
+export interface SemesterFilterState {
+  absence: {
+    type: string;
+  };
+}
+
+export interface FacultyFilterState {
+  area: string;
+  lastName: string;
+  firstName: string;
+  category: string;
+  fall: SemesterFilterState;
+  spring: SemesterFilterState;
+}
```

Comment:
Ah that's a good point - thanks for explaining that Robert. I updated these to be types instead and have them defined in the `FacultyScheduleTable` instead of the `FacultyPage` in order to get around the dependency cycle eslint error that I was encountering.